src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -42,7 +42,6 @@
   /** Cache of type variable/argument pairs */
   private static final Map<Class<?>, Reference<Map<TypeVariable<?>, Type>>> typeVariableCache = Collections
       .synchronizedMap(new WeakHashMap<Class<?>, Reference<Map<TypeVariable<?>, Type>>>());
-  private static volatile int METHOD_REF_OFFSET = -1;
```

Comment:
On my system, this always was `3`, so we weren't really saving much work by caching it. As a simplification, I figured it's safe to just leave this out since it seems cheap to compute and because we seem to need to keep the looping logic anyways in order to handle the `<init>` and `valueOf` special-cases mentioned below.